shared/actions/signup/index.js

```diff
@@ -242,8 +242,12 @@ function signup (skipMail): TypedAsyncAction<Signup | ShowPaperKey> {
   })
 }
 
-export function resetSignup (): ResetSignup {
-  return {type: Constants.resetSignup, payload: {}}
+export function resetSignup (): TypedAsyncAction<ResetSignup | RouteAppend> {
+  return dispatch => new Promise((resolve, reject) => {
+    dispatch({type: Constants.resetSignup, payload: {}})
+    dispatch(nextPhase())
```

Comment:
Maybe it should be called `updateRouter`.

I don't understand how it would be implied?

One option is to have the router reducer listen to this action, but that's a whole bunch of other problems that we've moved away from.

The other option is to fire an action that, itself, changes the router. That's what this is.

We could also not use the router, and then we wouldn't need to call this since components can just look at the `phase` (like how this works: https://github.com/keybase/client/pull/2316).

But as long as we have a router, we'll need to do something that updates the router's state.